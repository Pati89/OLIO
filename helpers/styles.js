// here we can add theme, fontSizes or spacing sizes to use same all over the page

export const breakpoints = {
  sm: 640,
  md: 896,
  lg: 1092,
};

export const media = {
  /* small size and up */
  sm: `@media only screen and (min-width: ${breakpoints.xs}px)`,
  /* medium size and up */
  md: `@media only screen and (min-width: ${breakpoints.sm}px)`,
  /* large size and up */
  lg: `@media (min-width: ${breakpoints.md}px)`,
};
