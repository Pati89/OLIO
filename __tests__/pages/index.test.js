import Home from "../../pages/index";
import { render } from "../../helpers/test-utils";

describe("Home", () => {
  it("should render a Home page", () => {
    const { getByTestId } = render(<Home />);
    expect(getByTestId("articles")).toBeTruthy();
  });

  it("should render no articles", () => {
    const { getByText } = render(<Home />);
    expect(getByText("No Articles to show")).toBeTruthy();
  });

  it("should render a Home match snapshot", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
