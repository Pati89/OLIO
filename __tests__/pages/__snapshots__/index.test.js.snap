// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render a Home match snapshot 1`] = `
.emotion-0 {
  margin: 0 20px;
}

@media only screen and (min-width: 640px) {
  .emotion-0 {
    margin: 0 50px;
  }
}

@media (min-width: 896px) {
  .emotion-0 {
    margin: 0 auto;
    max-width: 900px;
  }
}

.emotion-1 {
  min-height: 100vh;
  padding: 50px 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  margin: 0;
  padding-bottom: 50px;
  line-height: 1.15;
  font-size: 3rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 2rem 0;
  border-top: 1px solid #eaeaea;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  font-weight: bold;
}

<div>
  <div
    class="emotion-0"
    data-testid="articles"
  >
    <main
      class="emotion-1"
    >
      <h1
        class="emotion-2"
      >
        New Articles
      </h1>
      <p>
        No Articles to show
      </p>
    </main>
    <footer
      class="emotion-3"
      data-testid="footer"
    >
      <p>
        Developed by 
        <span
          class="emotion-4"
        >
          Patrycja Banaszczyk
        </span>
      </p>
    </footer>
  </div>
</div>
`;
